.github/workflows/build-nightly-ci-docker-images.yml

```diff
@@ -0,0 +1,75 @@
+name: Build docker images (Nightly CI)
+
+on:
+  workflow_call:
+  push:
+    branches:
+      - build_nightly_ci_docker_image*
+
+concurrency:
+  group: docker-images-builds
+  cancel-in-progress: false
+
+jobs:
+  latest-with-torch-nightly-docker:
+    name: "Nightly PyTorch + Stable TensorFlow"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cleanup disk
+        run: |
+          sudo ls -l /usr/local/lib/
+          sudo ls -l /usr/share/
+          sudo du -sh /usr/local/lib/
+          sudo du -sh /usr/share/
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /usr/share/dotnet
+          sudo du -sh /usr/local/lib/
+          sudo du -sh /usr/share/
```

Comment:
That is quite a lot of cleanup, where is this coming from?

Reply:
Disk full error occurred  when I tried to install torch 2.0.0 (pre) while the official version is not out yet. It's probably no longer necessary after the official version is released.

- Not really a lot of cleanup: there are only `2`. Other commands are just to print information.
- The only useful cleanup is `/usr/local/lib/android` which is about 12 GB.
- (The disk full error is due to torch was trying to find/install several versions that would match some version requirements)

Reply:
Is this important to keep? Every line of code in yml file is going to be copy/pasted across several others as work is done in these files/as new files pop up, so I'd be wary of adding non-necessary commands in the files :)

Reply:
@LysandreJik  Unfortunately, we still have the disk full issue. (See this [job run page](https://github.com/huggingface/transformers/actions/runs/4566445144))


<img width="695" alt="Screenshot 2023-03-30 182830" src="https://user-images.githubusercontent.com/2521628/228903122-516771f6-315e-491a-9565-f6f5435c82e7.png">

Reply:
BTW, this is already  `.github/workflows/build-docker-images.yml` on the `main` branch. Here is just the same code for the new file `.github/workflows/build-nightly-ci-docker-images.yml`.

Reply:
Sounds good, let's keep it then!